pkg/controllers/configuration.go

```diff
@@ -171,15 +203,38 @@ func (r *HypershiftDeploymentReconciler) ensureConfiguration(ctx context.Context
 		}
 
 		for _, se := range secretRefs {
-			k := genKey(se, hyd)
-			t, err := r.generateSecret(ctx, k, overrideNamespace(helper.GetHostingNamespace(hyd)))
+			// 1. Use user provided secret
+			k := genKey(se.secretRef, hyd)
+			secret, err := r.generateSecret(ctx, k, overrideNamespace(helper.GetHostingNamespace(hyd)))
 			if err != nil {
-				r.Log.Error(err, fmt.Sprintf("failed to copy secret %s", k))
+				r.Log.Info(fmt.Sprintf("failed to find and copy secret %s: %s", k, err.Error()))
```

Comment:
With Info the log message should be something like "did not find and copy secret.." if its informational. To bad there is no warning level.